core/nodejsActionBase/runner.js

```diff
@@ -67,8 +67,7 @@ function NodeActionRunner() {
         } else {
             // The code is a plain old JS file.
             try {
-                eval(message.code);
-                thisRunner.userScriptMain = eval(message.main);
+                thisRunner.userScriptMain = eval('(function(){\n' + message.code + '\nreturn ' + message.main + '})()');
```

Comment:
another advantage of using a file is that the file can be named appropriately. the net is that any resulting stack traces from invocation of the action will show up with myAction.js:32

whereas with eval, stack traces will include the runner.js file name and line numbers, with anonymous:32 as the nested file:line